src/main/java/de/tum/in/www1/artemis/domain/Result.java

```diff
@@ -218,14 +219,15 @@ public Result hasFeedback(Boolean hasFeedback) {
     }
 
     /**
-     * 1. set score 2. set successful = true, if score >= 100 or false if not
+     * 1. set score and round it to 4 decimal places
+     * 2. set successful = true, if score >= 100 or false if not
      *
      * @param score new score
      */
     public void setScore(Double score) {
         if (score != null) {
-            this.score = score;
-            this.successful = score >= 100.0;
+            this.score = roundToFourDecimalPlaces(score);
```

Comment:
Can you please add a comment here on why rounding happens here? I saw the comment in the function itself but why is this an issue only here. It might be confusing for developers reading the code, as the score of a result is not rounded any more in other places of the codebase.

Reply:
yep you're right. I will add a comment 👍

Reply:
done in 20cd2a8efb4c9630ecd1bfc55392b3ca37e9076c